io.github.wivrn.wivrn.yml

```diff
@@ -0,0 +1,209 @@
+      - -DFETCHCONTENT_BASE_DIR=deps
+      - -DXRT_IPC_MSG_SOCK_FILENAME=wivrn/comp_ipc
+      - -Wno-dev
+    sources:
+      - type: git
+        url: https://github.com/WiVRn/WiVRn.git
+        tag: 6da658bcfb4f67abd6c697f2c222529a20ac10ca # v0.18 + patches for flatpak not yet merged in master
+      - type: archive
+        url: https://github.com/boostorg/pfr/archive/refs/tags/2.2.0.tar.gz
+        dest: deps/boostpfr-src
+        sha256: 6f490ec1383de5855dcf801c23f0a967f888eb6e1440b6ca0bf36c12d4ae7ac7
+      - type: git
+        url: https://gitlab.freedesktop.org/monado/monado
+        tag: dfc602288ab05131584a3f2be18031a13fccd061
+        dest: deps/monado-src
+      - type: shell
+        commands:
+         - cd deps/monado-src && ../../patches/apply.sh ../../patches/monado
+    build-commands:
+      - install -DTm644 images/wivrn.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.wivrn.wivrn.svg
+      - install -Dm644 flatpak/io.github.wivrn.wivrn.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo
+      - install -Dm644 flatpak/io.github.wivrn.wivrn.desktop -t ${FLATPAK_DEST}/share/applications
+
+  - name: opencomposite
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DUSE_SYSTEM_OPENXR=OFF
+      - -DUSE_SYSTEM_GLM=OFF
+    sources:
+      - type: git
+        url: https://gitlab.com/znixian/OpenOVR.git
+        tag: 632e5cc50b913e93194ca2970e6f13021182579f
+      - type: patch
+        paths:
+          - opencomposite-0001-Add-install-target.patch
+
+  - name: launcher
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 launch-wivrn-server.sh ${FLATPAK_DEST}/bin/launch-wivrn-server
+    sources:
+      - type: script
+        dest-filename: launch-wivrn-server.sh
+        commands:
+        - rm -f $XDG_CONFIG_HOME/openvr/openvrpaths.vrpath
+        - sed -ne 's#app-path=\(.*\)#{"runtime":["\1/OpenComposite"],"version":1}#p' /.flatpak-info > $XDG_CONFIG_HOME/openvr/openvrpaths.vrpath
+        - chmod -w $XDG_CONFIG_HOME/openvr/openvrpaths.vrpath
+        - wivrn-server
+
+cleanup:
+  - /lib/pkgconfig
+  - /share/pkgconfig
+  - /include
+  - /share/man
+  - /share/cmake
+  - /share/gir-1.0
+  - /lib/girepository-1.0
+  - /lib/systemd
+  - '*.a'
+  - '*.la'
+
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
+  # GPU acceleration
+  - --device=dri
+  # Needs to talk to the network
+  - --share=network
+  # Announce WiVRn via avahi
+  - --system-talk-name=org.freedesktop.Avahi
+  # Allow getting the pretty hostname
+  - --system-talk-name=org.freedesktop.hostname1
+  # - --socket=system-bus # workaround bug between sd-bus (https://github.com/systemd/systemd/issues/16610) and xdg-dbus-proxy (https://github.com/flatpak/xdg-dbus-proxy/issues/21)
+  # Pipewire
+  - --filesystem=xdg-run/pipewire-0
+  # Set the active OpenXR / OpenVR runtime
+  - --filesystem=xdg-config/openxr:create
+  - --filesystem=xdg-config/openvr:create
```

Comment:
Why is do you need to touch host's config?

Reply:
The application is an OpenXR + OpenVR runtime, to be used by other applications.

For OpenVR, other applications can discover it by reading `$XDG_CONFIG_HOME/openvr/openvrpaths.vrpath`, then dlopen the library located in `.runtime[0]` + `bin/linux64/vrclient.so`.

For OpenXR, the process is described in https://registry.khronos.org/OpenXR/specs/1.0/loader.html#runtime-interaction. `$XDG_CONFIG_HOME/openxr/1/active_runtime.json` is a json document, where `.runtime.library_path` is the library to dlopen.

We "publish" our runtime in those directories, for others to read, which is why it has to be visible by the host.